internal/utils/cluster.go

```diff
@@ -71,3 +71,22 @@ func FindName(realNames []string, name string) string {
 	log.Error().Msgf("Error: no real name found for %s", name)
 	return ""
 }
+
+// ExtractTargetPorts extracts target ports defined inside the role in the LoadBalancer.
+func ExtractTargetPorts(loadBalancers []*pb.LBcluster) []int {
+	ports := make(map[int32]struct{})
+
+	for _, c := range loadBalancers {
+		for _, role := range c.Roles {
+			ports[role.TargetPort] = struct{}{}
```

Comment:
Can we do something like 
```go
if _, ok := ports[role.TargetPort]; !ok {
	result = append(result, int(role.TargetPort))
	ports[role.TargetPort] = struct{}{}
}
```
thus omitting the second `for` loop

Reply:
Yes, that's a more simpler way